frontend/src/components/global/AlgoTextarea.vue

```diff
@@ -18,6 +26,41 @@ export default {
             this.$nextTick(() => {
                 event.target.selectionStart = event.target.selectionEnd = selectionStart + 1;
             });
+
+            this.$emit("change-specific", {
+                start: event.target.selectionStart,
+                end: event.target.selectionStart + 1,
+                size: 1,
+                deltaLineCount: 0,
+            });
+        },
+
+        handleBeforeinput(event) {
+            oldText = event.target.value;
+        },
+
+        handleInput(event) {
+            let newText = event.target.value;
+
+            let oldLen = oldText.length;
+            let newLen = newText.length;
+
+            let lengthDifference = newLen - oldLen;
+            let differenceStart = 0;
+
+            for (let i = 0; i < Math.min(newLen, oldLen); i++) {
+                if (newText[i] == oldText[i]) differenceStart = i + 1;
+                else break;
+            }
+
+            let oldLineCount = (oldText.match(/\n/g) || []).length;
+            let newLineCount = (newText.match(/\n/g) || []).length;
+            this.$emit("change-specific", {
+                start: differenceStart,
+                end: differenceStart + Math.abs(lengthDifference),
```

Comment:
Zastanawiam się, czy nie lepiej end traktować jako ostatni różniący się znak w kodzie. Jak podmienisz fragment kodu o długości x na zupełnie inny fragment kodu o tej samej długości, no to start == end, size = 0 czyli tak jakby zmiany nie było.